From c3765bc1333fbe89370f0a8b86a1ab8fe06678fa Mon Sep 17 00:00:00 2001
From: Karthik T S <quic_kartikts@quicinc.com>
Date: Thu, 8 Aug 2024 10:56:50 +0530
Subject: [PATCH] net core: Add default disable config for fast recycled skbs
 debug

Add default disable config for fast recycled skbs debug

Change-Id: I7cd85aa9609810b184d4d333483c7468fcee40fa
Signed-off-by: Neha Bisht <quic_nbisht@quicinc.com>
Signed-off-by: Karthik T S <quic_kartikts@quicinc.com>
---
 include/linux/skbuff.h | 2 +-
 net/Kconfig            | 7 +++++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index a7eabac5354f..e1fc874f580e 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -1422,7 +1422,7 @@ static inline int skb_pad(struct sk_buff *skb, int pad)
 }
 #define dev_kfree_skb(a)	consume_skb(a)
 #define dev_kfree_skb_list_fast(a)	consume_skb_list_fast(a)
-#if defined(SKB_FAST_RECYCLABLE_DEBUG_ENABLE) && defined(CONFIG_SKB_RECYCLER)
+#if defined(CONFIG_SKB_FAST_RECYCLABLE_DEBUG_ENABLE)
 #define dev_check_skb_fast_recyclable(a)       check_skb_fast_recyclable(a)
 #else
 #define dev_check_skb_fast_recyclable(a)
diff --git a/net/Kconfig b/net/Kconfig
index 53754fd2fe9a..47b8114f649b 100644
--- a/net/Kconfig
+++ b/net/Kconfig
@@ -419,6 +419,13 @@ config SKB_TIMESTAMP
 	bool "Timestamp storage enablement in the packet"
 	default n
 
+config SKB_FAST_RECYCLABLE_DEBUG_ENABLE
+	bool "Enable debug API for fast recycled skbs"
+	depends on SKB_RECYCLER
+	default n
+	help
+	 Enable debug API for fast recycled skbs.
+
 menu "Network testing"
 
 config NET_PKTGEN
-- 
2.34.1

